main.go

```diff
@@ -113,53 +112,31 @@ func main() {
 					pinClient := pinclient.NewClient(endpoint, c.String(tokenFlag.Name)) // instantiate client with token
 
 					// Create libp2p host
-					host, err := libp2p.New(
-						libp2p.NATPortMap(),
-						libp2p.EnableHolePunching(),
-					)
+					host, err := libp2p.New()
```

Comment:
You make a good point here that some pinning services might not provide delegates https://github.com/2color/auspinner/commit/6e2473cd192626f81fcdcf681694323112902ce2#r76506860.

IIUC the spec has origins as optional and delegates as required, but that the pinning service can still try and fetch the data from elsewhere even if the delegates/origins aren't quite working (bc content addressing means you may find it elsewhere 😄).

I wouldn't want you to have to wait an extra 10 seconds just to pin a 10kb JSON blob though. Some options include:
1. Just drop the port mapping and hole punching
2. Have a flag to enable it that's off by default
3. Setup everything with port mapping and hole punching and do a pin without waiting on addresses or relays, then if your addresses change do a pin replace adding in the new origins.
    - This should be something pinning services can handle... but you never know
    - Even if you do this I'd put a flag in (that makes this not the default) either way because you just know someone's going to complain about "but.... FTP doesn't need to make connections to relay servers just to upload data how come IPFS eats up resources 😭" even though we're just trying to give them more options for a scenario where FTP would fail.

Reply:
I took the approach you suggest in the third option only to discover that updating pin requests with the same CID in order to pass the origins doesn't work.

So my current thinking is to add a flag to conditionally wait and pass the origins in the request.